pkg/ratelimiter/rate_limiter.go

```diff
@@ -0,0 +1,67 @@
+package ratelimiter
+
+import (
+	"math"
+	"sync"
+	"time"
+)
+
+var (
+	DefaultBaseDelay = 2 * time.Minute
+	DefaultMaxDelay  = 10 * time.Minute
+	defaultMaxTries  = 3
+)
+
+type ItemExponentialFailureRateLimiterWithMaxTries struct {
+	// failuresLock mutex, which is necessary due to the fact that controller-runtime allows for concurrent reconciler invocations for a single controller
+	failuresLock sync.Mutex
+	failures     map[interface{}]int
+	maxTries     int
+
+	baseDelay time.Duration
+	maxDelay  time.Duration
+}
+
+func NewItemExponentialFailureRateLimiterWithMaxTries(baseDelay time.Duration, maxDelay time.Duration) RateLimiter {
+	return &ItemExponentialFailureRateLimiterWithMaxTries{
+		failures:  map[interface{}]int{},
+		baseDelay: baseDelay,
+		maxDelay:  maxDelay,
+		maxTries:  defaultMaxTries,
+	}
+}
+
+func (r *ItemExponentialFailureRateLimiterWithMaxTries) When(item interface{}) time.Duration {
+	r.failuresLock.Lock()
+	defer r.failuresLock.Unlock()
+
+	r.failures[item] = r.failures[item] + 1
+
+	if r.failures[item] > r.maxTries {
+		// reminder
+		return 60 * time.Minute
+	}
+
+	backoff := float64(r.baseDelay.Nanoseconds()) * math.Pow(2, float64(r.failures[item]))
+	durationBackoff := time.Duration(backoff)
+
+	if durationBackoff > r.maxDelay {
+		return r.maxDelay
+	}
```

Comment:
It seems `if durationBackoff > r.maxDelay` condition never happens. 
By default, there is a 2-minute delay for the first reconciliation `backoff = 2min * 2^0 = 2min`, for the second there is a 4-minute delay `2min * 2^1 = 4min`, for the third - an 8-minute delay `2min * 2^2 = 8min`. 
For the 4th reconcile and so on there is a 60-minute delay.
This statement only works if there are more than 3 reconcile retries so it would be better to make it configurable.